pkg/util/cloudfoundry/types_test.go

```diff
@@ -238,3 +279,32 @@ func TestGetVcapServicesMap(t *testing.T) {
 	assert.Nil(t, err)
 	assert.EqualValues(t, expected, result)
 }
+
+func TestIsAllowedTag(t *testing.T) {
+	// when both empty, exclude everything
+	includeList := []*regexp.Regexp{}
+	excludeList := []*regexp.Regexp{}
+
+	result := isAllowedTag("aRandomValue", includeList, excludeList)
+	assert.EqualValues(t, false, result)
+
+	// include strings in the includeList and not in the excludeList
+	includeList = []*regexp.Regexp{regexp.MustCompile("include.*")}
+	excludeList = []*regexp.Regexp{}
+
+	result = isAllowedTag("includeTag", includeList, excludeList)
+	assert.EqualValues(t, true, result)
+
+	result = isAllowedTag("excludeTag", includeList, excludeList)
+	assert.EqualValues(t, false, result)
+
+	// reject strings in the excludeList
+	includeList = []*regexp.Regexp{}
+	excludeList = []*regexp.Regexp{regexp.MustCompile("exclude.*")}
+
+	result = isAllowedTag("aRandomValue", includeList, excludeList)
+	assert.EqualValues(t, true, result)
+
+	result = isAllowedTag("excludeTag", includeList, excludeList)
+	assert.EqualValues(t, false, result)
```

Comment:
let's just add a bit more complex case, where there is an intersection between whitelist and blacklist, to make sure the blacklist precedence is respected